render-tests/runtime-styling/set-style-layer-change-source-type/style.json

```diff
@@ -0,0 +1,74 @@
+{
+  "version": 8,
+  "metadata": {
+    "ignored": {
+      "native": "https://github.com/mapbox/mapbox-gl-native/issues/2445",
+      "js": "https://github.com/mapbox/mapbox-gl-js/issues/3633"
+    },
+    "test": {
+      "width": 64,
+      "height": 64,
+      "operations": [
+        [
+          "setStyle",
+          {
+            "version": 8,
+            "transition": {
+              "duration": 0
+            },
+            "sources": {
+              "geojson": {
+                "type": "geojson",
+                "data": {
+                  "type": "Point",
+                  "coordinates": [
+                    0,
+                    0
+                  ]
+                }
+              }
+            },
+            "sprite": "local://sprites/sprite",
+            "layers": [
+              {
+                "id": "circle",
+                "type": "circle",
+                "source": "geojson"
+              }
```

Comment:
Reason that `expected.png` looks right: the original `circle` layer uses an sdf icon colored red.